### PR TITLE
feat: implement PEP249 Connections and Cursor API

### PR DESCRIPTION
**Pull request by dingbo8128**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,17 +34,13 @@ import taosrest
 conn = taosrest.connect(host="localhost",
                         user="root",
                         password="taosdata",
-                        database="test",
                         prot=6041)
```
Comment (sangshuduo) [inline README.md:37]: port

Comment (sangshuduo) [reply README.md:37]: no timeout

File: taosrest/__init__.py [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
-def connect():
-    pass
+
+def connect(**kwargs) -> TaosRestConnection:
+    """
+   Keyword Arguments
+   ----------------------------
+   - host : str, optional.
+       host to connect
+   - port : int, optional.
```
Comment (sangshuduo) [inline taosrest/__init__.py:10]: arguments order should align with README

File: taosrest/connection.py [hunk 3]
```diff
@@ -1,9 +1,34 @@
...
+    Implement [PEP 249 Connection API](https://peps.python.org/pep-0249/#connection-objects)
+    """
+
     def __init__(self, **kwargs):
+        """
+        Keyword Arguments
+        ----------------------------
+        - host : str, optional.
+            host to connect
+        - port : int, optional.
```
Comment (sangshuduo) [inline taosrest/connection.py:17]: should align with README

File: tests/test_restful_connection.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import taosrest
+
+
+def test_fetch_all():
+    conn = taosrest.connect(host="localhost",
+                            user="root",
+                            password="taosdata",
+                            database="test",
+                            prot=6041)
```
Comment (sangshuduo) [inline tests/test_restful_connection.py:9]: port

File: tests/test_restful_connection.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        print(row)
+    print(cursor.description)
+
+
+def test_fetch_one():
+    conn = taosrest.connect(host="localhost",
+                            user="root",
+                            password="taosdata",
+                            database="test",
+                            prot=6041)
```
Comment (sangshuduo) [inline tests/test_restful_connection.py:24]: port

**Review by sangshuduo — CHANGES_REQUESTED** [review]
see my comments

File: README.md (hunk 1 above)
Comment (dingbo8128) [reply README.md:37]: leaving timeout as default is good for most cases.

Comment (dingbo8128) [reply README.md:37]: api referenc will explain all parameters.  this example not need.

File: taosrest/__init__.py (hunk 2 above)
Comment (dingbo8128) [reply taosrest/__init__.py:10]: ok

File: taosrest/connection.py (hunk 3 above)
Comment (dingbo8128) [reply taosrest/connection.py:17]: ok

**Review by sangshuduo — APPROVED** [review]
LGTM

**Review by zitsen — APPROVED** [review]
LGTM

**Pull request merged**